azure-pipelines.yml

```diff
@@ -202,17 +213,28 @@ stages:
         displayName: '.Net UTs'
 
       - task: PublishPipelineArtifact@1
+        condition: always()
```

Comment:
Can you also fail `Java UTs` and `Java ITs` assertions to make sure they are published too when failed?
We might need this trick on more places.

Reply:
..and maybe give them also the `testRunTitle` if possible (in an easy way)

Reply:
For the Java UTs, we have
```yaml
      - task: Maven@3
        displayName: 'Maven verify, UTs with SonarCloud'
        inputs:
          goals: 'verify'
          options: -B --settings $(mavenSettings.secureFilePath) -Pcoverage -Dsonar.projectVersion=$(SONAR_PROJECT_VERSION)
          publishJUnitResults: true
          testResultsFiles: '**/surefire-reports/TEST-*.xml'
          testRunTitle: 'UTs on Windows'
          javaHomeOption: 'JDKVersion'
          jdkVersionOption: '1.11'
          mavenOptions: $(MAVEN_OPTS)
          sonarQubeRunAnalysis: true
          sqMavenPluginVersionChoice: 'latest'

      - task: SonarCloudPublish@1
        displayName: 'Code Analysis - Publish QG'
        inputs:
          pollingTimeoutSec: '300'
```

We have `publishJUnitResults: true`, which does the push to Azure DevOps. We could add `condition: always()` to the `SonarCloudPublish@1` task. I don't know how I can test this. I would need to fail one of the Java UTs. Can we do such things in another PR?

Reply:
You can just break any assertion, just like you did on .NET side.

UTs - for example expect 42 instead: 
https://github.com/SonarSource/sonar-dotnet/blob/c827aa62dbfeb2a3a3e0f6ad428323910e7b69e9/sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java#L55

ITs - for example expect "BROKEN IT":
https://github.com/SonarSource/sonar-dotnet/blob/c827aa62dbfeb2a3a3e0f6ad428323910e7b69e9/its/src/test/java/com/sonar/it/csharp/AnalysisWarningsTest.java#L62